### ISSUE TITLE
Bug in thermal_speed?

### ISSUE THREAD

**Issue by pheuer** [PlasmaPy/PlasmaPy#1051]
I think I'm getting a bug in `plasmapy.formulary.parameters.thermal_speed`: I get a RuntimeWarning from decorator validator for what I think should be perfectly acceptable input. Minimal example below

Example:
```
import astropy.units as u
import warnings

from plasmapy.formulary.parameters import thermal_speed

# Display warning as error for traceback
warnings.simplefilter("error")

Te = 116045 * u.K # Equivalent to 10 eV
vTe = thermal_speed(Te, particle="e")
```

**Comment by StanczakDominik** [comment]
I can't reproduce that on my machine; could you provide the version of astropy and of plasmapy (probably the commit hash) where you see this?

**Comment by StanczakDominik** [comment]
whoops!

**Comment by pheuer** [comment]
I'm on the master branch (I think the commit hash is `af7b9d3') and `astropy==4.0.1.post1'

I still get the warning when I upgrade to `astropy=4.2.0`

**Comment by pheuer** [comment]
The warning does disappear if I roll back to `v5.0` (`9a253db`)

**Comment by StanczakDominik** [comment]
Okay, that's weird. Can I ask for a traceback copypaste?

**Comment by pheuer** [comment]
Sure, here it is

```
Traceback (most recent call last):

  File "E:\Dropbox\Documents\Research\PlasmaPy\2021_2_28 Bug in thermal_speed.py", line 18, in <module>
    vTe = thermal_speed(Te, particle="e")

  File "c:\users\peter\documents\github\plasmapy\plasmapy\utils\decorators\helpers.py", line 137, in wrapper
    return f(*args, **kwargs)

  File "c:\users\peter\documents\github\plasmapy\plasmapy\utils\decorators\checks.py", line 1318, in wrapper
    return_ = f(*args, **kwargs)

  File "c:\users\peter\documents\github\plasmapy\plasmapy\utils\decorators\helpers.py", line 137, in wrapper
    return f(*args, **kwargs)

  File "c:\users\peter\documents\github\plasmapy\plasmapy\utils\decorators\validators.py", line 199, in wrapper
    bound_args.arguments[arg_name], arg_name, validations[arg_name]

  File "c:\users\peter\documents\github\plasmapy\plasmapy\utils\decorators\validators.py", line 374, in _validate_quantity
    self._check_value(arg, arg_name, arg_validations)

  File "c:\users\peter\documents\github\plasmapy\plasmapy\utils\decorators\checks.py", line 319, in _check_value
    if not arg_checks[ckey] and np.any(arg < 0):

  File "C:\Users\Peter\Anaconda3\envs\plasmapy\lib\site-packages\astropy\units\quantity.py", line 477, in __array_ufunc__
    result = super().__array_ufunc__(function, method, *arrays, **kwargs)

RuntimeWarning: invalid value encountered in less
```

**Comment by StanczakDominik** [comment]
Nope, still can't reproduce! Since you're running this through conda, could you share the output of `conda list` (from the anaconda terminal), and maybe even `pip freeze` for good measure? At this point I think this is some weird numpy/python combination issue...

**Comment by pheuer** [comment]
Weird...

My environment is pretty messy, so maybe I just need to clean it up...

`conda list`

```
# packages in environment at C:\Users\Peter\Anaconda3\envs\plasmapy:
#
# Name                    Version                   Build  Channel
alabaster                 0.7.12                   py37_0
appdirs                   1.4.4                    pypi_0    pypi
argh                      0.26.2                   py37_0
asteval                   0.9.18                   pypi_0    pypi
astroid                   2.4.1                    py37_0
astropy                   4.0.1.post1              pypi_0    pypi
atomicwrites              1.4.0                      py_0
attrs                     19.3.0                     py_0
autopep8                  1.4.4                      py_0
babel                     2.8.0                      py_0
backcall                  0.1.0                    py37_0
bcrypt                    3.1.7            py37he774522_0
black                     20.8b1                   pypi_0    pypi
black-nbconvert           0.2.0                    pypi_0    pypi
blas                      1.0                         mkl
bleach                    3.1.5                    pypi_0    pypi
brotli                    1.0.9                ha925a31_2
brotlipy                  0.7.0           py37he774522_1000    anaconda
ca-certificates           2020.10.14                    0    anaconda
cached-property           1.5.2                    pypi_0    pypi
certifi                   2020.6.20                py37_0    anaconda
cffi                      1.14.0           py37h7a1dbc1_0
cfgv                      3.2.0                    pypi_0    pypi
chardet                   3.0.4                 py37_1003
click                     7.1.2                    pypi_0    pypi
cloudpickle               1.4.1                      py_0
colorama                  0.4.3                      py_0    conda-forge
corner                    2.1.0                    pypi_0    pypi
cryptography              2.9.2            py37h7a1dbc1_0
cycler                    0.10.0                     py_2    conda-forge
decorator                 4.4.2                      py_0
defusedxml                0.6.0                      py_0
diff-match-patch          20181111                   py_0
distlib                   0.3.1                    pypi_0    pypi
docutils                  0.16                     py37_0
emcee                     3.0.2                    pypi_0    pypi
entrypoints               0.3                      pypi_0    pypi
filelock                  3.0.12                   pypi_0    pypi
flake8                    3.8.4                      py_0
freetype                  2.10.2               hd328e21_0    conda-forge
future                    0.18.2                   py37_0
h5py                      2.10.0          nompi_py37h422b98e_102    conda-forge
hdf5                      1.10.5          nompi_ha405e13_1104    conda-forge
hypothesis                5.11.0                     py_0
icc_rt                    2019.0.0             h0cc432a_1
icu                       64.2                 he025d50_1    conda-forge
identify                  1.4.29                   pypi_0    pypi
idna                      2.9                        py_1
imageio                   2.9.0                    pypi_0    pypi
imagesize                 1.2.0                      py_0
importlib-metadata        1.6.0                    py37_0
importlib_metadata        1.6.0                         0
intel-openmp              2019.4                      245
intervaltree              3.0.2                      py_0
ipykernel                 5.3.0                    pypi_0    pypi
ipython                   7.15.0                   pypi_0    pypi
ipython_genutils          0.2.0                    py37_0
ipywidgets                7.5.1                    pypi_0    pypi
isort                     4.3.21                   py37_0
jedi                      0.17.0                   pypi_0    pypi
jinja2                    2.11.2                     py_0
joblib                    1.0.0                    pypi_0    pypi
jpeg                      9d                   he774522_0    conda-forge
jsonschema                3.2.0                    py37_0
jupyter_client            6.1.3                      py_0
jupyter_core              4.6.3                    py37_0
keyring                   21.1.1                   py37_2
kiwisolver                1.2.0            py37heaa310e_0    conda-forge
lazy-object-proxy         1.4.3            py37he774522_0
libblas                   3.8.0                    14_mkl    conda-forge
libcblas                  3.8.0                    14_mkl    conda-forge
libclang                  9.0.1           default_hf44288c_0    conda-forge
liblapack                 3.8.0                    14_mkl    conda-forge
libpng                    1.6.37               hfe6a214_1    conda-forge
libsodium                 1.0.16               h9d3ae62_0
libspatialindex           1.9.3                h33f27b4_0
libtiff                   4.1.0                h885aae3_6    conda-forge
llvmlite                  0.34.0                   pypi_0    pypi
lmfit                     1.0.1                    pypi_0    pypi
lz4-c                     1.9.2                h62dcd97_1    conda-forge
m2w64-gcc-libgfortran     5.3.0                         6
m2w64-gcc-libs            5.3.0                         7
m2w64-gcc-libs-core       5.3.0                         7
m2w64-gmp                 6.1.0                         2
m2w64-libwinpthread-git   5.0.0.4634.697f757               2
magpylib                  2.3.0b0                  pypi_0    pypi
markupsafe                1.1.1            py37he774522_0
matplotlib                3.2.1                         0    conda-forge
matplotlib-base           3.2.1            py37h911224e_0    conda-forge
mccabe                    0.6.1                    py37_1
meshio                    4.1.1                    pypi_0    pypi
mistune                   0.8.4            py37he774522_0
mkl                       2019.4                      245
mkl-service               2.3.0            py37hfa6e2cd_0    conda-forge
more-itertools            8.3.0                      py_0
mpmath                    1.1.0                      py_0    conda-forge
msys2-conda-epoch         20160418                      1
mypy-extensions           0.4.3                    pypi_0    pypi
nbconvert                 5.6.1                    py37_0
nbformat                  5.0.6                      py_0
nbsphinx                  0.7.0                    pypi_0    pypi
nodeenv                   1.5.0                    pypi_0    pypi
notebook                  6.0.3                    py37_0
numba                     0.51.2                   pypi_0    pypi
numdifftools              0.9.39                   pypi_0    pypi
numpy                     1.18.4           py37hae9e721_0    conda-forge
numpydoc                  1.0.0                    pypi_0    pypi
olefile                   0.46                       py_0    conda-forge
openssl                   1.1.1h               he774522_0    anaconda
packaging                 20.4                     pypi_0    pypi
pandas                    1.1.4                    pypi_0    pypi
pandoc                    2.2.3.2                       0
pandocfilters             1.4.2                    py37_1
paramiko                  2.7.1                      py_0
parso                     0.7.0                      py_0
pathspec                  0.8.1                    pypi_0    pypi
pathtools                 0.1.2                      py_1
pexpect                   4.8.0                    py37_0
pickleshare               0.7.5                    py37_0
pillow                    7.1.2            py37hc826c6e_0    conda-forge
pip                       20.1.1                     py_1    conda-forge
plasmapy                  0.3.2.dev33+gc5020b6           dev_0    <develop>
pluggy                    0.13.1                   py37_0
pradformat                0.1.0                    pypi_0    pypi
pradreader                1.0a2                    pypi_0    pypi
praline                   0.0.6                     dev_0    <develop>
pre-commit                2.10.1                   pypi_0    pypi
prns                      1.0                       dev_0    <develop>
prometheus-client         0.8.0                    pypi_0    pypi
prometheus_client         0.7.1                      py_0
prompt-toolkit            3.0.5                    pypi_0    pypi
prompt_toolkit            3.0.4                         0
psutil                    5.7.0            py37he774522_0
py                        1.8.1                      py_0
pycodestyle               2.6.0                      py_0
pycparser                 2.20                       py_0
pydocstyle                4.0.1                      py_0
pyerfa                    1.7.2                    pypi_0    pypi
pyflakes                  2.2.0                      py_0
pygments                  2.7.0                    pypi_0    pypi
pylint                    2.5.2                    py37_0
pynacl                    1.3.0            py37h62dcd97_0
pyopenssl                 19.1.0                   py37_0
pyparsing                 2.4.7              pyh9f0ad1d_0    conda-forge
pyqt                      5.12.3           py37h1834ac0_3    conda-forge
pyqt5-sip                 4.19.18                  pypi_0    pypi
pyqtchart                 5.12                     pypi_0    pypi
pyqtwebengine             5.12.1                   pypi_0    pypi
pyreadline                2.1                   py37_1001    conda-forge
pyrsistent                0.16.0           py37he774522_0
pysocks                   1.7.1                    py37_0
pytest                    5.4.3                    pypi_0    pypi
pytest-arraydiff          0.3              py37h39e3cac_0
pytest-astropy            0.8.0                      py_0
pytest-astropy-header     0.1.2                      py_0
pytest-doctestplus        0.5.0                      py_0
pytest-openfiles          0.5.0                      py_0
pytest-remotedata         0.3.2                    py37_0
python                    3.7.6           cpython_h60c2a47_6    conda-forge
python-dateutil           2.8.1                      py_0    conda-forge
python-jsonrpc-server     0.3.4                      py_0
python-language-server    0.34.1                   py37_0
python_abi                3.7                     1_cp37m    conda-forge
pytz                      2020.1                     py_0
pyvista                   0.25.3                   pypi_0    pypi
pywin32                   227                      pypi_0    pypi
pywin32-ctypes            0.2.0                 py37_1000
pywinpty                  0.5.7                    py37_0
pyyaml                    5.3.1            py37he774522_0
pyzmq                     19.0.1                   pypi_0    pypi
qdarkstyle                2.8.1                      py_0
qt                        5.12.5               h7ef1ec2_0    conda-forge
qtawesome                 0.7.0                      py_0
qtconsole                 4.7.4                      py_0
qtpy                      1.9.0                      py_0
regex                     2020.11.13               pypi_0    pypi
requests                  2.23.0                   py37_0
rope                      0.17.0                     py_0
rtree                     0.9.4            py37h21ff451_1
scikit-learn              0.24.1                   pypi_0    pypi
scipy                     1.3.1            py37h29ff71c_0    conda-forge
scooby                    0.5.6                    pypi_0    pypi
send2trash                1.5.0                    py37_0
setuptools                47.1.1           py37hc8dfbb8_0    conda-forge
setuptools-scm            4.1.2                    pypi_0    pypi
six                       1.15.0             pyh9f0ad1d_0    conda-forge
snowballstemmer           2.0.0                      py_0
sortedcontainers          2.1.0                    py37_0
sphinx                    2.4.4                    pypi_0    pypi
sphinx-automodapi         0.12                     pypi_0    pypi
sphinx-copybutton         0.2.11                   pypi_0    pypi
sphinx-gallery            0.7.0                    pypi_0    pypi
sphinx-rtd-theme          0.4.3                    pypi_0    pypi
sphinxcontrib-applehelp   1.0.2                      py_0
sphinxcontrib-devhelp     1.0.2                      py_0
sphinxcontrib-htmlhelp    1.0.3                      py_0
sphinxcontrib-jsmath      1.0.1                      py_0
sphinxcontrib-qthelp      1.0.3                      py_0
sphinxcontrib-serializinghtml 1.1.4                      py_0
spyder                    4.1.5                    py37_0
spyder-kernels            1.9.4                    py37_0
sqlite                    3.30.1               hfa6e2cd_0    conda-forge
sunbear                   0.1.0                    pypi_0    pypi
sympy                     1.6.1                    pypi_0    pypi
terminado                 0.8.3                    pypi_0    pypi
testpath                  0.4.4                      py_0
threadpoolctl             2.1.0                    pypi_0    pypi
tk                        8.6.10               hfa6e2cd_0    conda-forge
toml                      0.10.2                   pypi_0    pypi
tornado                   6.0.4            py37hfa6e2cd_0    conda-forge
tqdm                      4.48.2                   pypi_0    pypi
traitlets                 4.3.3                    py37_0
typed-ast                 1.4.1            py37he774522_0
typing-extensions         3.7.4.3                  pypi_0    pypi
ujson                     1.35             py37hfa6e2cd_0
uncertainties             3.1.2                    pypi_0    pypi
urllib3                   1.25.9                   pypi_0    pypi
vc                        14.1                 h869be7e_1    conda-forge
virtualenv                20.0.31                  pypi_0    pypi
vs2015_runtime            14.16.27012          h30e32a0_2    conda-forge
vtk                       9.0.1                    pypi_0    pypi
watchdog                  0.10.2                   py37_0
wcwidth                   0.2.3                    pypi_0    pypi
webencodings              0.5.1                    pypi_0    pypi
wheel                     0.34.2                     py_1    conda-forge
widgetsnbextension        3.5.1                    pypi_0    pypi
win_inet_pton             1.1.0                    py37_0
wincertstore              0.2                   py37_1003    conda-forge
winpty                    0.4.3                         4
wrapt                     1.11.2           py37he774522_0
xarray                    0.17.0                   pypi_0    pypi
xz                        5.2.5                h2fa13f4_0    conda-forge
yaml                      0.1.7                hc54c509_2
yapf                      0.28.0                     py_0
zeromq                    4.3.1                h33f27b4_3
zipp                      3.1.0                      py_0
zlib                      1.2.11            h2fa13f4_1006    conda-forge
zstd                      1.4.4                h9f78265_3    conda-forge
```

`pip freeze`

```
alabaster==0.7.12
appdirs==1.4.4
argh==0.26.2
asteval==0.9.18
astroid==2.4.1
astropy==4.2
atomicwrites==1.4.0
attrs==19.3.0
autopep8==1.4.4
Babel==2.8.0
backcall==0.1.0
bcrypt==3.1.7
black==20.8b1
black-nbconvert==0.2.0
bleach==3.1.5
brotlipy==0.7.0
cached-property==1.5.2
certifi==2020.6.20
cffi==1.14.0
cfgv==3.2.0
chardet==3.0.4
click==7.1.2
cloudpickle==1.4.1
colorama==0.4.3
corner==2.1.0
cryptography==2.9.2
cycler==0.10.0
decorator==4.4.2
defusedxml==0.6.0
diff-match-patch==20181111
distlib==0.3.1
docutils==0.16
emcee==3.0.2
entrypoints==0.3
filelock==3.0.12
flake8 @ file:///tmp/build/80754af9/flake8_1601911421857/work
future==0.18.2
h5py==2.10.0
hypothesis==5.11.0
identify==1.4.29
idna==2.9
imageio==2.9.0
imagesize==1.2.0
importlib-metadata==1.6.0
intervaltree==3.0.2
ipykernel==5.3.0
ipython==7.15.0
ipython-genutils==0.2.0
ipywidgets==7.5.1
isort==4.3.21
jedi @ file:///C:/ci/jedi_1592833832328/work
Jinja2==2.11.2
joblib==1.0.0
jsonschema==3.2.0
jupyter-client==6.1.3
jupyter-core==4.6.3
keyring==21.1.1
kiwisolver==1.2.0
lazy-object-proxy==1.4.3
llvmlite==0.34.0
lmfit==1.0.1
magpylib==2.3.0b0
MarkupSafe==1.1.1
matplotlib==3.2.1
mccabe==0.6.1
meshio==4.1.1
mistune==0.8.4
mkl-service==2.3.0
more-itertools==8.3.0
mpmath==1.1.0
mypy-extensions==0.4.3
nbconvert==5.6.1
nbformat==5.0.6
nbsphinx==0.7.0
nodeenv==1.5.0
notebook==6.0.3
numba==0.51.2
numdifftools==0.9.39
numpy @ file:///D:/bld/numpy_1588605187257/work
numpydoc==1.0.0
olefile==0.46
packaging==20.4
pandas==1.1.4
pandocfilters==1.4.2
paramiko==2.7.1
parso==0.7.0
pathspec==0.8.1
pathtools==0.1.2
pexpect==4.8.0
pickleshare==0.7.5
Pillow==7.1.2
-e git+https://github.com/pheuer/PlasmaPy.git@ed8f1c5f60be3b4d3c6222f7667743c7e77a7cbc#egg=plasmapy
pluggy==0.13.1
pradformat @ git+https://github.com/phyzicist/pradformat.git@40745b125f124833ed12cc9c7dcaa6915f259c0c#subdirectory=Python
pradreader @ git+https://github.com/flash-center/PRadReader@b63c7c778bc6ade914141b66e1ba87fbbaf68e64
# Editable install with no version control (praline==0.0.6)
-e c:\users\peter\documents\flash\praline
pre-commit==2.10.1
-e git+https://github.com/OxfordHED/proton-radiography-no-source/@9ead3048a8cbbe6aa10879e763182f7693230e6c#egg=prns
prometheus-client==0.8.0
prompt-toolkit==3.0.5
psutil==5.7.0
py==1.8.1
pycodestyle==2.6.0
pycparser==2.20
pydocstyle==4.0.1
pyerfa==1.7.2
pyflakes==2.2.0
Pygments==2.7.0
pylint==2.5.2
PyNaCl==1.3.0
pyOpenSSL==19.1.0
pyparsing==2.4.7
PyQt5==5.12.3
PyQt5-sip==4.19.18
PyQtChart==5.12
PyQtWebEngine==5.12.1
pyreadline==2.1
pyrsistent==0.16.0
PySocks==1.7.1
pytest==5.4.3
pytest-arraydiff==0.3
pytest-astropy==0.8.0
pytest-astropy-header==0.1.2
pytest-doctestplus==0.5.0
pytest-openfiles==0.5.0
pytest-remotedata==0.3.2
python-dateutil==2.8.1
python-jsonrpc-server==0.3.4
python-language-server @ file:///C:/ci/python-language-server_1594152769556/work
pytz==2020.1
pyvista==0.25.3
pywin32==227
pywin32-ctypes==0.2.0
pywinpty==0.5.7
PyYAML==5.3.1
pyzmq==19.0.1
QDarkStyle==2.8.1
QtAwesome==0.7.0
qtconsole==4.7.4
QtPy==1.9.0
regex==2020.11.13
requests==2.23.0
rope==0.17.0
Rtree==0.9.4
scikit-learn==0.24.1
scipy==1.3.1
scooby==0.5.6
Send2Trash==1.5.0
setuptools-scm==4.1.2
six @ file:///home/conda/feedstock_root/build_artifacts/six_1590081179328/work
snowballstemmer==2.0.0
sortedcontainers==2.1.0
Sphinx==3.0.3
sphinx-automodapi==0.12
sphinx-copybutton==0.2.11
sphinx-gallery==0.7.0
sphinx-rtd-theme==0.4.3
sphinxcontrib-applehelp==1.0.2
sphinxcontrib-devhelp==1.0.2
sphinxcontrib-htmlhelp==1.0.3
sphinxcontrib-jsmath==1.0.1
sphinxcontrib-qthelp==1.0.3
sphinxcontrib-serializinghtml==1.1.4
spyder @ file:///C:/ci/spyder_1599052032150/work
spyder-kernels @ file:///C:/ci/spyder-kernels_1599051798778/work
sunbear==0.1.0
sympy==1.6.1
terminado==0.8.3
testpath==0.4.4
threadpoolctl==2.1.0
toml==0.10.2
tornado==6.0.4
tqdm==4.48.2
traitlets==4.3.3
typed-ast==1.4.1
typing-extensions==3.7.4.3
ujson==1.35
uncertainties==3.1.2
urllib3 @ file:///tmp/build/80754af9/urllib3_1603305693037/work
virtualenv==20.0.31
vtk==9.0.1
watchdog==0.10.2
wcwidth==0.2.3
webencodings==0.5.1
widgetsnbextension==3.5.1
win-inet-pton==1.1.0
wincertstore==0.2
wrapt==1.11.2
xarray==0.17.0
yapf==0.28.0
zipp==3.1.0
```

**Comment by StanczakDominik** [comment]
Nope, I'm at a loss :(

**Comment by pheuer** [comment]
@StanczakDominik I _think_ I've solved the issue by upgrading my numpy from 1.18.4 to 1.19.2. Did you try lowering your numpy to the minimal version? 

If that doesn't doesn't do it, then we can just close this (since it's solved on my PC and apparently unreproducible elsewhere!)

**Comment by rocco8773** [comment]
From where the error is occurring, it looks like it would be either an `astropy` or `numpy` issue.  I did set my environment to have `astropy==4.0.1.post1` and `numpy==1.18.4` and was not able to reproduce the error.

On a side note, you can now use `eV` for temperature and no warnings will be issued.

---

I will close this issue for now, but feel free to ping if the issue comes back.
